game/lib/model/event_model.dart

```diff
@@ -11,12 +11,24 @@ class EventModel extends ChangeNotifier {
 
   EventModel(ApiClient client) : _client = client {
     client.clientApi.updateEventDataStream.listen((event) {
-      if (!event.event is String) {
-        searchResults = event.event;
+      if (!event.deleted) {
+        //initialize searchResults if null
+        searchResults = searchResults ?? [];
+        //add or update event
+        if (!_events.containsKey(event.event.id)) {
+          searchResults!.add(event.event);
```

Comment:
The issue with this condition is if we request event data outside the search functionality it will suddenly appear there (which is not the case at the moment but this could cause a bug in the future).

Reply:
You can fix this without touching the API at all by adding filtering inside the model. For example, if you have a filter in the frontend (arts quad only + non expiring + whatever), you can set it inside the model whenever searchEvents is called (along with the requestAllEvents call). On every _events update, rerun the filter over all the events and put all those inside searchResults.

Reply:
Shouldn't the filtering be done on the widget page? Because otherwise based on the filtering the user does, the filter function in event_model will also change